R/zchunk_L210.water.demand.primary.R

```diff
@@ -34,26 +34,57 @@ module_water_L210.water.demand.primary_DISABLED <- function(command, ...) {
     A22.globaltech_coef <- get_data(all_data, "energy/A22.globaltech_coef")
     L110.water_demand_primary_R_S_W_m3_GJ <- get_data(all_data, "L110.water_demand_primary_R_S_W_m3_GJ")
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are `merge` calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+      # ===================================================
+      # TRANSLATED PROCESSING CODE GOES HERE...
+      #
+      A21.globaltech_coef %>% select(supplysector,subsector,technology) -> A21.tmp1
+      A22.globaltech_coef %>% filter( grepl('nuclear', supplysector)) %>% select(supplysector,subsector,technology) -> A22.tmp
+      A21.A22.tmp <- bind_rows(A21.tmp1,A22.tmp)
+      left_join(L110.water_demand_primary_R_S_W_m3_GJ, A21.A22.tmp, by="supplysector") -> L210.TechCoef.v1
+
+
+      # Avoid double acounting unconventional oil in the regional oil sector as it is already
+      # accounted for in unconventional oil production.
+      dontuse1 <- tibble(supp=c("regional oil"))
+      dontuse2 <- tibble(subs=c("unconventional oil"))
+
+      L210.TechCoef.v1  %>%
+        filter(!(supplysector %in% dontuse1$supp & subsector %in% dontuse2$subs)) %>%
+        mutate(water_sector = "Mining") %>%
+        mutate(minicam.energy.input = set_water_input_name(water_sector, water_type, A03.sector)) %>%
+        # Add in GCAM region names
+        left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+        mutate(market.name=region) -> L210.TechCoef.v2
+
+      L210.orig_num_rows <- nrow( L210.TechCoef.v2 )
+      L210.TechCoef.v2[rep( 1:L210.orig_num_rows, times=length( MODEL_YEARS ) ),] %>%
+        mutate(year = MODEL_YEARS[sort(rep(1:length(MODEL_YEARS), times=L210.orig_num_rows))]) %>%
+        select( one_of(names_TechCoef) ) -> results
```

Comment:
What's going on in these lines? Very unclear. Need more/better comments, at least.